src/uiScreens.c

```diff
@@ -21,6 +21,60 @@ void ui_displayPathScreen(
 	);
 }
 
+// the given path typically corresponds to an account
+// if it contains anything more, we display just the whole path
+void ui_displayAccountScreen(
+        const char* screenHeader,
+        const bip44_path_t* path,
+        ui_callback_fn_t callback
+)
+{
+	ASSERT(strlen(screenHeader) > 0);
+	ASSERT(strlen(screenHeader) < BUFFER_SIZE_PARANOIA);
+
+	ASSERT(bip44_hasValidCardanoPrefix(path));
+	ASSERT(bip44_containsAccount(path));
+
+	char accountDescription[160];
+	explicit_bzero(accountDescription, SIZEOF(accountDescription));
+
+	if (bip44_hasReasonableAccount(path) && !bip44_containsChainType(path)) {
```

Comment:
the `bip44_containsChainType` check technically does the check for the path length but it's imho not that obvious for a casual reader of the code. I would rather extend the bip44 library with a function analogous for the `bip44_containsMoreThanAddress` but for Accounts - this would imho make the purpose of the check clear at first sight